pkg/provisioner/ironic/clients/client.go

```diff
@@ -19,15 +19,23 @@ var tlsConnectionTimeout = time.Second * 30
 // TLSConfig contains the TLS configuration for the Ironic connection.
 // Using Go default values for this will result in no additional trusted
 // CA certificates and a secure connection.
+// When specifying Certificate and Private key, TLS connection will use
+// client certificate authentication.
 type TLSConfig struct {
-	TrustedCAFile      string
-	InsecureSkipVerify bool
+	TrustedCAFile         string
+	ClientCertificateFile string
+	ClientPrivateKeyFile  string
+	InsecureSkipVerify    bool
+	SkipClientSANVerify   bool
```

Comment:
Is this used ?

Reply:
ClientCertificateFile, ClientPrivateKeyFile is used
InsecureSkipVerify was before this PR
SkipClientSANVerify not used, but inspired by InsecureSkipVerify, as also could be useful

Reply:
Sorry, I only meant SkipClientSANVerify. if you think that this is useful, then it should probably be exposed through an environment variable, no ?

Reply:
hmm, yea, valid point, I somehow missed that InsecureSkipVerify is also exposed via envars, thanks

Reply:
@maelk added IRONIC_SKIP_CLIENT_VERIFY environ in latest PR rebase